fairyfishgui.py

```diff
@@ -321,13 +323,26 @@ def update_board(self, variant=None, fen=None, move=None, undo=False):
         if move:
             self.board.state.push(move)
             if self.engine:
-                self.engine.position(self.board.state.start_fen, self.board.state.moves)
+                self.engine.position(
+                    self.board.state.start_fen, self.board.state.moves)
         if undo:
             self.board.state.pop()
             if self.engine:
                 self.engine.position(self.board.state.start_fen, self.board.state.moves)
+				
+        if nnue:
```

Comment:
Setting the options should be independent from updating the board state, so I would suggest to either use `self.engine.setoption` directly from within the main loop or wrap it into another method (`self.set_engine_option` or the like). Stopping and restarting the engine search as part of that makes sense, but resetting the position seems unnecessary to me.